volatility/plugins/mftparser.py

```diff
@@ -139,39 +140,66 @@ class MFT_FILE_RECORD(obj.CType):
+    # For deleted files, this does not match but thats better than showing active files with incorrect paths.
+    # Also added workaround for handling recently deleted files in get_full_path() function below. -YK
     def add_path(self, fileinfo):
         # it doesn't really make sense to add regular files to parent directory,
         # since they wouldn't actually be in the middle of a file path, but at the end
         # therefore, we'll return for regular files
-        if not self.is_directory():
-            return
+        # # Removing this optimization because we need the files for continuing MFT records 
+        #   which reference baseRecord. - YK
+        #if not self.is_directory(): 
+        #    return
         # otherwise keep a record of the directory that we've found
-        cur = MFT_PATHS_FULL.get(int(self.RecordNumber), None)
+        ref = long(self.RecordNumber | (self.SequenceValue << 48))
+        cur = MFT_PATHS_FULL.get(ref, None) # 6 bytes of record, 2 bytes of seq, makes 8 bytes of reference
         if (cur == None or fileinfo.Namespace != 2) and fileinfo.is_valid():
             temp = {}
-            temp["ParentDirectory"] = fileinfo.ParentDirectory
+            temp["ParentDirectory"] = long(fileinfo.ParentDirectory)
             temp["filename"] = self.remove_unprintable(fileinfo.get_name())
-            MFT_PATHS_FULL[int(self.RecordNumber)] = temp
+            temp["deleted"] = not self.is_inuse()  # needed for validating a folder in case deleted
+            temp["directory"] = self.is_directory() 
+            MFT_PATHS_FULL[ref] = temp
 
     def get_full_path(self, fileinfo):
         if self.obj_vm._config.DEBUGOUT:
             print "Building path for file {0}".format(fileinfo.get_name())
         parent = ""
         path = self.remove_unprintable(fileinfo.get_name()) or "(Null)"
         try:
-            parent_id = fileinfo.ParentDirectory & 0xffffff
+            parent_id = long(fileinfo.ParentDirectory) # & 0xffffffffffff # Bug fixed but not needed now
         except struct.error:
             return path
-        if int(self.RecordNumber) == 5 or int(self.RecordNumber) == 0:
+        if long(self.RecordNumber) == 5 or long(self.RecordNumber) == 0:
             return path
         seen = set()
+        processing_deleted = False
         while parent != {}:
-            seen.add(parent_id)
-            parent = MFT_PATHS_FULL.get(int(parent_id), {})
-            if parent == {} or parent["filename"] == "" or int(parent_id) == 0 or int(parent_id) == 5:
+            if processing_deleted == False : seen.add(parent_id)
+            parent = MFT_PATHS_FULL.get(parent_id, {})
+            if parent == {} and processing_deleted == False:
+                # Perhaps its parent folder is deleted and the parent folder's sequence number has advanced by one
+                rec = parent_id & 0xffffffffffff
+                seq = ( (parent_id >> 48) + 1 ) << 48 # incrementing seq by 1
```

Comment:
Please remove the extra spaces here:

`( (parent_id >> 48) + 1 )` should be `((parent_id >> 48) + 1)`